lib/mayaUsd/render/vp2ShaderFragments/UsdPreviewSurfaceLightShadowAPI2.xml

```diff
@@ -0,0 +1,111 @@
+<!--
+========================================================================
+Copyright 2018 Pixar
+Copyright 2020 Autodesk
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+========================================================================
+-->
+<fragment_graph name="UsdPreviewSurface" ref="UsdPreviewSurface" class="FragmentGraph" version="1.0" feature_level="0">
+    <fragments>
+        <!-- Fragments for extracting computed dot products. -->
+        <fragment_ref name="NwFaceCameraIfNAN" ref="NwFaceCameraIfNAN"/>
+
+        <!-- Fragments for computing material. -->
+        <fragment_ref name="opacityToTransparency" ref="opacityToTransparency"/>
+        <fragment_ref name="usdPreviewSurfaceLightingAPI2" ref="usdPreviewSurfaceLightingAPI2"/>
+
+        <!-- Maya fragments for computing the output surface values. -->
+        <fragment_ref name="maya_FogDepthSurface" ref="maya_FogDepthSurface"/>
+        <fragment_ref name="mayaComputeSurfaceFinal" ref="mayaComputeSurfaceFinal"/>
+        <fragment_ref name="mayaSurfaceShaderOutput" ref="mayaSurfaceShaderOutput"/>
+
+        <!-- Light effects fragment in case we want PCF shadows -->
+        <fragment_ref name="usdPreviewSurfaceShadowAPI2" ref="usdPreviewSurfaceShadowAPI2"/>
+    </fragments>
+    <connections>
+        <connect from="NwFaceCameraIfNAN.output" to="usdPreviewSurfaceLightingAPI2.Nw" name="Nw"/>
+        <connect from="opacityToTransparency.transparency" to="usdPreviewSurfaceLightingAPI2.transparency" name="transparency"/>
+        <connect from="usdPreviewSurfaceLightingAPI2.surfaceShader" to="mayaComputeSurfaceFinal.input" name="input"/>
+        <connect from="maya_FogDepthSurface.maya_FogDepthSurface" to="mayaComputeSurfaceFinal.maya_FogDepthSurface" name="maya_FogDepthSurface"/>
+        <connect from="mayaComputeSurfaceFinal.mayaComputeSurfaceFinal" to="usdPreviewSurfaceShadowAPI2.inSurface" name="usdPreviewSurfaceShadowAPI2"/>
```

Comment:
Adding usdPreviewSurfaceShadowAPI2 between the SurfaceFinal fragment and the output struct is the only difference between this new fragment and the previous one.